llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp

```diff
@@ -673,10 +673,33 @@ AArch64LegalizerInfo::AArch64LegalizerInfo(const AArch64Subtarget &ST)
 
   getActionDefinitionsBuilder({G_SITOFP, G_UITOFP})
       .legalForCartesianProduct({s32, s64, v2s64, v4s32, v2s32})
+      .legalIf([=](const LegalityQuery &Query) {
+        return HasFP16 &&
```

Comment:
```
typeInSet(0, {s16, v4s16, v8s16})(Query) && typeInSet(1, {s32, s64, v4s16, v8s16})(Query)
```
WDYT?

Reply:
Hello. This way at least feels clear to me what is going on and matches what we have elsewhere. We might be able to adjust them in the future to clean things up, but for the moment it's probably best to keep G_SITOFP similar to G_FPTOSI.

I'm interested to hear your suggestions if we can make the whole things clearer. Maybe it would be worth having a `.legalIf(HasFP16, {{s16, s32}, {s16, s64}, ...})`